napari_arboretum/graph.py

```diff
@@ -48,7 +49,7 @@ def build_reverse_graph(graph: dict) -> list:
         A reversed graph representing children of each parent node.
     """
     reverse_graph = {}
-    roots = set()
+    roots: Union[list, set] = set()
```

Comment:
similarly I'm not sure what the `list` contains here

I've add to do this because `roots` is initially a `set` (I assume to benefit from uniqueness) but then transforms into ` list` later on

Reply:
I think these are `int` again